ts/features/bonus/siciliaVola/saga/networking/handleGetListaComuniBySiglaProvincia.ts

```diff
@@ -0,0 +1,71 @@
+import { ActionType } from "typesafe-actions";
+import { call, put } from "redux-saga/effects";
+import { BackendSiciliaVolaClient } from "../../api/backendSiciliaVola";
+import { getGenericError, getNetworkError } from "../../../../../utils/errors";
+import { svGenerateVoucherAvailableMunicipality } from "../../store/actions/voucherGeneration";
+import { SagaCallReturnType } from "../../../../../types/utils";
+import { Municipality } from "../../types/SvVoucherRequest";
+import { isDefined } from "../../../../../utils/guards";
+import { ComuneBeanList } from "../../../../../../definitions/api_sicilia_vola/ComuneBeanList";
+
+// TODO: add the mapping when the swagger will be fixed
+const mapKinds: Record<number, string> = {};
+
+// convert a success response to the logical app representation of it
+const convertSuccess = (
+  listaComuni: ComuneBeanList
+): ReadonlyArray<Municipality> =>
+  listaComuni
+    .map(r =>
+      r.codiceCatastale && r.descrizioneComune
+        ? {
+            id: r.codiceCatastale,
+            name: r.descrizioneComune
+          }
+        : undefined
+    )
+    .filter(isDefined);
```

Comment:
```suggestion
    listaComuni.flatMap(r =>
        r.codiceCatastale && r.descrizioneComune
            ? [{
                id: r.codiceCatastale,
                name: r.descrizioneComune
              }]
            : []
    )
```

It will avoid parsing the collection twice.

Reply:
Oh great, I never used it before but is of course better!
Note that to implement this improvement I need to add the lib in [this commit](https://github.com/pagopa/io-app/pull/3267/commits/a16d20fb3cac5c62a83b12e550f1757c3056b592) to the `tsconfig`.

Refactored in [08e5d39](https://github.com/pagopa/io-app/pull/3267/commits/08e5d39ff9a542ed77f47de7345d406db1fc1e19)